lib/LedgerSMB/Scripts/import_csv.pm

```diff
@@ -428,32 +441,41 @@ our $process = {
     inventory_multi => \&_process_inventory_multi,
 };
 
-=head2 parse_file
+=head2 _parse_file
 
 This parses a file, and returns a the csv in tabular format.
 
 =cut
 
-sub parse_file {
+sub _parse_file {
     my $self = shift @_;
 
     my $handle = $self->{_request}->upload('import_file');
     my $contents = join("\n", <$handle>);
+    my $sep = "";
+    my $n = @{$cols->{$self->{type}}}-1; # 1 separator less than fields
 
     $self->{import_entries} = [];
     for my $line (split /(\r\n|\r|\n)/, $contents){
-        next if ($line !~ /,/);
+        if ( $sep eq "" ) {
+            if    ($n == (() = $line =~  /,/g)) { $sep =  "," }
+            elsif ($n == (() = $line =~  /;/g)) { $sep =  ";" }
+            elsif ($n == (() = $line =~ /\t/g)) { $sep = "\t" }
+            else  { die "Unrecognized file format" }
```

Comment:
It is called Comma Separated Values, and RFC 4180 was an attempt to standardize, it never made it globally. CSV has been abused by many vendors and export tools for Delimiter Based Fields (typically a single reserved character such as comma, semicolon, or tab; sometimes the delimiter may include optional spaces). See https://en.wikipedia.org/wiki/Comma-separated_values for a description.